src/audio/drc/drc_math_lib/db2mag/c_fixpt/drc_db2mag.c

```diff
@@ -0,0 +1,463 @@
+
+    for (i = 0; i < nm1; i++)
+    {
+    }
+
+    while (i < n)
+    {
+      y[i] = 0U;
+      i++;
+    }
+  }
+}
+
+/*
+ * Arguments    : const uint32_T u1[]
+ *                int32_T n1
+ *                const uint32_T u2[]
+ *                int32_T n2
+ *                uint32_T y[]
+ *                int32_T n
+ * Return Type  : void
+ */
+static void uMultiWordMul(const uint32_T u1[], int32_T n1, const uint32_T u2[],
+  int32_T n2, uint32_T y[], int32_T n)
+{
+  int32_T i;
+  uint32_T cb;
+  uint32_T u1i;
+  int32_T a1;
+  int32_T a0;
+  int32_T ni;
+  int32_T k;
+  int32_T j;
+  int32_T b1;
+  int32_T b0;
+  uint32_T w01;
+  uint32_T yk;
+  uint32_T t;
+
+  /* Initialize output to zero */
+  if (0 <= (n - 1))
+  {
+    memset(&y[0], 0, ((uint32_T)n) * (sizeof(uint32_T)));
+  }
+
+  for (i = 0; i < n1; i++)
+  {
+    cb = 0U;
+    u1i = u1[i];
+    a1 = (int32_T)((uint32_T)(u1i >> 16U));
+    a0 = (int32_T)((uint32_T)(u1i & 65535U));
+    ni = n - i;
+    if (n2 <= ni)
+    {
+      ni = n2;
+    }
+
+    k = i;
+    for (j = 0; j < ni; j++)
+    {
+      u1i = u2[j];
+      b1 = (int32_T)((uint32_T)(u1i >> 16U));
+      b0 = (int32_T)((uint32_T)(u1i & 65535U));
+      u1i = ((uint32_T)a1) * ((uint32_T)b0);
+      w01 = ((uint32_T)a0) * ((uint32_T)b1);
+      yk = y[k] + cb;
+      cb = (uint32_T)((yk < cb) ? 1 : 0);
+      t = ((uint32_T)a0) * ((uint32_T)b0);
+      yk += t;
+      cb += (uint32_T)((yk < t) ? 1 : 0);
+      t = (u1i << 16U);
+      yk += t;
+      cb += (uint32_T)((yk < t) ? 1 : 0);
+      t = (w01 << 16U);
+      yk += t;
+      cb += (uint32_T)((yk < t) ? 1 : 0);
+      y[k] = yk;
+      cb += (u1i >> 16U);
+      cb += (w01 >> 16U);
+      cb += ((uint32_T)a1) * ((uint32_T)b1);
+      k++;
+    }
+
+    if (k < n)
+    {
+      y[k] = cb;
+    }
+  }
+}
+
+/*
+ * Arguments    : const uint32_T u1[]
+ *                int32_T n1
+ *                uint32_T n2
+ *                uint32_T y[]
+ *                int32_T n
+ * Return Type  : void
+ */
+static void uMultiWordShr(const uint32_T u1[], int32_T n1, uint32_T n2, uint32_T
+  y[], int32_T n)
+{
+  int32_T nb;
+  int32_T i;
+  int32_T nc;
+  uint32_T nr;
+  int32_T i1;
+  uint32_T nl;
+  uint32_T u1i;
+  uint32_T yi;
+  nb = ((int32_T)n2) / 32;
+  i = 0;
+  if (nb < n1)
+  {
+    nc = n + nb;
+    if (nc > n1)
+    {
+      nc = n1;
+    }
+
+    nr = n2 - (((uint32_T)nb) * 32U);
+    if (nr > 0U)
+    {
+      nl = 32U - nr;
+      u1i = u1[nb];
+      for (i1 = nb + 1; i1 < nc; i1++)
+      {
+        yi = (u1i >> nr);
+        u1i = u1[i1];
+        y[i] = yi | (u1i << nl);
+        i++;
+      }
+
+      yi = (u1i >> nr);
+      if (nc < n1)
+      {
+        yi |= (u1[nc] << nl);
+      }
+
+      y[i] = yi;
+      i++;
+    }
+    else
+    {
+      for (i1 = nb; i1 < nc; i1++)
+      {
+        y[i] = u1[i1];
+        i++;
+      }
+    }
+  }
+
+  while (i < n)
+  {
+    y[i] = 0U;
+    i++;
+  }
+}
+
+/*
+ * function y = drc_db2mag_fixpt(tstruct)
```

Comment:
No description about algorithm, comments like SOF src/math/decibels.c math would be nice to understand how this algorithm works.